### PR TITLE
Added private_only setting, as well as another method of adding userdata

### PR DESCRIPTION
**Pull request by kyletbase2**

### CODE REVIEW

File: tests/userdata.test.yaml [hunk 1]
```diff
@@ -1,8 +1,10 @@
...
   name: userdata
-  description: add extra userdata
+  description: Userdata as a list with newlines
 
-userdata: |
-  /usr/bin/bootstrap
-  echo "userdata complete"
+userdata:
+  - "cd /tmp \n"
+  - "sudo yum install -y https://s3.amazonaws.com/ec2-downloads-windows/SSMAgent/latest/linux_amd64/amazon-ssm-agent.rpm \n"
```
Comment (aaronwalker) [inline tests/userdata.test.yaml:8]: Did you test this with a AMI that already had SSM agent installed?

Comment (aaronwalker) [reply tests/userdata.test.yaml:8]: should we just include this in the standard userdata?

**Review by aaronwalker — COMMENTED** [review]
other than that it looks good

File: tests/userdata.test.yaml (hunk 1 above)
Comment (kyletbase2) [reply tests/userdata.test.yaml:8]: @aaronwalker 
I did test it with Amazon Linux 2, it just updated the binary, BUT, as of writing this, only the regional link works:
https://s3.ap-southeast-2.amazonaws.com/amazon-ssm-ap-southeast-2/latest/linux_amd64/amazon-ssm-agent.rpm

This link doesn't (times out):
https://s3.amazonaws.com/ec2-downloads-windows/SSMAgent/latest/linux_amd64/amazon-ssm-agent.rpm
In the cloud-init-output.log:
```
Cannot open: https://s3.amazonaws.com/ec2-downloads-windows/SSMAgent/latest/linux_amd64/amazon-ssm-agent.rpm. Skipping.
Error: Nothing to do
```
But it kept going with the rest of the userdata script.

IMO it should be included, as well as maybe changing the default IAMs to include what I included with this customer:
```
  iam_policies:
    ecs-ssm:
      action:
        - ssmmessages:*
        - ssm:*
        - s3:GetEncryptionConfiguration
```

**Review by Guslington — APPROVED** [review]

**Pull request merged**